src/csharp/SDK/Capture.cs

```diff
@@ -169,98 +157,235 @@ public Image Color
-                    if (disposedValue)
-                        throw new ObjectDisposedException(nameof(Capture));
+                    if (this.disposedValue || other.disposedValue)
+                    {
+                        return false;
+                    }
 
-                    NativeMethods.k4a_capture_set_temperature_c(handle, value);
+                    IntPtr myHandleValue = this.handle.DangerousGetHandle();
+                    IntPtr otherHandleValue = other.handle.DangerousGetHandle();
+
+                    // If both images represent the same native handle, consider them equal
+                    return myHandleValue == otherHandleValue;
                 }
             }
         }
 
-        public Capture Reference()
+        /// <summary>
+        /// Handle the disposing of the object.
+        /// </summary>
+        /// <param name="disposing">true when called by Dispose(), false when called by the finalizer.</param>
+        protected virtual void Dispose(bool disposing)
         {
             lock (this)
             {
-                if (disposedValue)
-                    throw new ObjectDisposedException(nameof(Capture));
+                if (!this.disposedValue && disposing)
+                {
+                    this.cachedColor?.Dispose();
+                    this.cachedDepth?.Dispose();
+                    this.cachedIR?.Dispose();
+
+                    this.cachedColor = null;
+                    this.cachedDepth = null;
+                    this.cachedIR = null;
+
+                    this.handle.Close();
+                    this.handle = null;
 
-                return new Capture(handle.DuplicateReference());
+                    Allocator.Singleton.UnregisterForDisposal(this);
+                }
+
+                this.disposedValue = true;
             }
         }
 
-        private NativeMethods.k4a_capture_t handle;
-
-        #region IDisposable Support
-        private bool disposedValue = false; // To detect redundant calls
-    
-        // This code added to correctly implement the disposable pattern.
-        public void Dispose()
+        /// <summary>
+        /// Retrieves a native image handle from the native API.
+        /// </summary>
+        /// <param name="nativeMethod">Native method to retrieve the image.</param>
+        /// <param name="cachedImage">A cached instance of the Image that we return to callers. (Callers may dispose this image, although they shouldn't).</param>
+        /// <remarks>
+        /// If this is the first time calling the property, we construct a new wrapper.
+        /// If the handle is for an Image we have already constructed a wrapper for, we return the existing wrapper.
+        /// If the handle is for a different Image, we construct a new wrapper and dispose the old one.
+        /// If existing wrapper has been disposed, we throw an exception.
+        /// </remarks>
+        private void GetImageWrapperAndDisposePrevious(
```

Comment:
Maybe rename to UpdateCachedImage or something like that since it isn't returning an image anymore?